lib/colorado_lottery.rb

```diff
@@ -0,0 +1,31 @@
+class ColoradoLottery
+  attr_reader :registered_contestants, :winners, :current_contestants
+
+  def initialize
+    @registered_contestants= {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(contestant, game)
+    if contestant.game_interests.include?(game.name) && contestant.age > 18
+      true
+    else
+      false
+    end
+  end
+
+  def can_register?(contestant, game)
+    if interested_and_18?(contestant, game) == true || contestant.out_of_state? == false
```

Comment:
not quite-- the contestant needed to be interested_and_18 _and_ not out of state _or_the lottery needed to be a national drawing.  So your || should be an &&, and you're missing another condition for checking the national drawing. 

```
  def can_register?(contestant, game)
    state_resident = !contestant.out_of_state?
    national_game = game.national_drawing? == true

    interested_and_18?(contestant, game) && (state_resident || national_game)
  end
```